### PR TITLE
[SPARK-36724][SQL] Support timestamp_ntz as a type of time column for SessionWindow

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR proposes to support `timestamp_ntz` as a type of time column for `SessionWIndow` like `TimeWindow` does.

### Why are the changes needed?

For better usability.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

New test.

### CODE REVIEW

**Review by HeartSaVioR — APPROVED** [review]
+1 Looks good to me.

**Comment by HeartSaVioR** [comment]
cc. @gengliangwang @viirya @xuanyuanking

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SessionWindow.scala [hunk 1]
```diff
@@ -69,10 +69,10 @@ case class SessionWindow(timeColumn: Expression, gapDuration: Expression) extend
   with NonSQLExpression {
 
   override def children: Seq[Expression] = Seq(timeColumn, gapDuration)
-  override def inputTypes: Seq[AbstractDataType] = Seq(TimestampType, AnyDataType)
+  override def inputTypes: Seq[AbstractDataType] = Seq(AnyTimestampType, AnyDataType)
   override def dataType: DataType = new StructType()
-    .add(StructField("start", TimestampType))
-    .add(StructField("end", TimestampType))
+    .add(StructField("start", children.head.dataType))
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SessionWindow.scala:74]: nit: `children.head.dataType` => `timeColumn.dataType`

**Review by gengliangwang — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**